### DISCUSSION TITLE
Need unmorphed user data from API

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Ed/Community#350] [Sunbird ED configuration discussion] [answered]
Hello All,

http://docs.sunbird.org/latest/apis/userapi/#operation/User_Search_V3
This is the API that gives user data but user mail is morphed --> k1**@yopmail.com
I want it to be unmorphed --> [k123@yopmail.com](mailto:k123@yopmail.com)
Is there any API that can give me that ?

Please guide me here .

cc: @komalm @Renuka-S

**Comment by Krishnaj20** [comment, marked as answer]
@reshmi-nair @rhwarrier - can you respond to this query?

**Reply by reshmi-nair** [reply]
@Xitija Currently there is no API which returns the unmasked value. There is a /private/user/v1/read/:uid API which provides the encrypted value which can be called from service to service. This encrypted value has to be decrypted to get the actual value.

**Reply by Xitija** [reply]
Is there any other way I can get this data?
Or any development that can be done to achieve this ?

cc: @komalm @Renuka-S

**Reply by pranavirabatti** [reply]
@Xitija I remember that I had shared the ticket with your team. In case you missed it I'm sending it here again. Please check this out. It can be helpful.

https://github.com/project-sunbird/sunbird-community/discussions/14

**Reply by Krishnaj20** [reply]
@Xitija @renuka-S - Please check with @coolbung as well. He had a similar query which was answered by @SanthoshVasabhaktula in-person. Also, if your query is answered request you to proactively mark them as answered - we are a small community will help in effectively managing active threads.

**Reply by manojLondhe** [reply]
@reshmi-nair  
- We got encrypted values using API `/private/user/v1/read/:uid`, and tried passing those to decrypt API `http://localhost:8013/decrypt` (from encryption service using https://github.com/project-sunbird/enc-service).
- But `http://localhost:8013/decrypt` API needs data to be passed in a format like 
```
{
  "value":"1.0.0|97|PKCS1|hNw8ClTKZ6KMYG5JlFngu5SNogBsdXa5iigO2k6r29hZ90vW+uKhPO1uvTIOp6i1eocCJxdt8QlVydsHD+x2Ew4O1RybiB2zfF7bxcasrGEC6zIV+L04mtLHIpST03LxrAen5bJyOMDuH9bx27CMnk0ut33bjMcw0dceL9vl5z+teG/rZZYihn5bAP6Rnvfh3qimp1JsddBmchI/OdTpm8VovOEICAqOa5DLGcxUTMrKEXyCtNefvcRaHLvVOF7pfYAkfrGZE6V24OdD0RzNHwuVBt/EHMsHIbdnsghzpwWCyXJ1QNu147krFFKv1Rq3N31zgkItdSh1FONrbptdoA=="
}
```
- and we only get ```hNw8ClTKZ6KMYG5JlFngu5SNogBsdXa5iigO2k6r29hZ90vW+uKhPO1uvTIOp6i1eocCJxdt8QlVydsHD+x2Ew4O1RybiB2zfF7bxcasrGEC6zIV+L04mtLHIpST03LxrAen5bJyOMDuH9bx27CMnk0ut33bjMcw0dceL9vl5z+teG/rZZYihn5bAP6Rnvfh3qimp1JsddBmchI/OdTpm8VovOEICAqOa5DLGcxUTMrKEXyCtNefvcRaHLvVOF7pfYAkfrGZE6V24OdD0RzNHwuVBt/EHMsHIbdnsghzpwWCyXJ1QNu147krFFKv1Rq3N31zgkItdSh1FONrbptdoA==``` part of the data from `/private/user/v1/read/:uid` API.

Questions: 
-- How do we get encrypted record id part of it eg: `97` in this case? 
-- Why is it necessary to decrypt?

@Krishnaj20 
As suggested by @SanthoshVasabhaktula to @coolbung, we tried looking at data-products code for decrypt, but could not figure it out yet.
https://github.com/Sunbird-Ed/sunbird-data-products/blob/release-5.0.0/data-products/src/main/scala/org/sunbird/analytics/util/DecryptUtil.scala

**Reply by Krishnaj20** [reply]
@manojLondhe - check user personal info exhaust at https://lern.sunbird.org/learn/product-and-developer-guide/batch-service/reports

**Reply by reshmi-nair** [reply]
@manojLondhe 
-- How do we get encrypted record id part of it eg: 97 in this case? 
While using enc-service if its not returning the decrypted data correctly, that means some configurations are missing. So pls recheck the enc-service configurations. You need to use the same sunbird_encryption_key used to encrypt the value while decrypting also. 

-- Why is it necessary to decrypt?
Decrypt is necessary if you want to see the data in plain text. All the PII information will be stored in encrypted format.

In data-products - DecryptUtil.scala code also, if you just set the same sunbird_encryption_key used for encryption as environement variable, this will work.

**Reply by manojLondhe** [reply]
thanks, will double check.

I meant why record id eg: 97 in this case, is needed for decrypting?

**Reply by reshmi-nair** [reply]
@manojLondhe "97" in the above decrypt request is the id of the key that was used for encryption. Without that the encryption service cannot figure out the key which was used for encryption.

**Reply by Krishnaj20** [reply]
@manojLondhe - is your query resolved?

**Reply by Krishnaj20** [reply]
@manojLondhe @Xitija - Hope the response shared by @reshmi-nair answers your query. Marking this as answered.
